ports/libmagic/portfile.cmake

```diff
@@ -31,11 +31,17 @@ file(COPY "${CMAKE_CURRENT_LIST_DIR}/unofficial-libmagic-config.cmake.in" DESTIN
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/magic.def" DESTINATION "${SOURCE_PATH}/src")
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/config.h" DESTINATION "${SOURCE_PATH}/src")
 
+if(VCPKG_CROSSCOMPILING)
+    vcpkg_add_to_path(PREPEND "${CURRENT_HOST_INSTALLED_DIR}/tools/libmagic")
+elseif(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+    set(EXTRA_ARGS "ADD_BIN_TO_PATH")
+endif()
```

Comment:
We move this block before the configure and install steps, so that libmagic's build system can find the `file` command in the path we built it for the host.

Reply:
Well, not quite sure what this does, but have you tested this with cross compilation? (If it works for x86_64 and crosscompilation x86_64->aarch64)

Reply:
It's tested as outlined in https://github.com/TileDB-Inc/TileDB/pull/5338#issuecomment-2424143758.

As for what this does, in libmagic's `CMakeLists.txt` we have this command:

```cmake
add_custom_command(OUTPUT magic.mgc
  COMMAND file -C -m magic
  COMMENT "Compiling magic file"
)
```

When not cross-compiling, CMake resolves `file` to the `file` executable we built, but when cross-compiling `file` will be passed to the shell as-is and resolved from `PATH`. The `vcpkg_add_to_path` command ensures the host's `file` we built gets found.

The `elseif` part makes sure `libmagic.dll` gets found on Windows if we are using dynamic linkage for the dependencies (which we don't currently do).

This block exists in the upstream port BTW.

https://github.com/microsoft/vcpkg/blob/b8d688b979bd54f0269ea1c6101e6a977613175f/ports/libmagic/portfile.cmake#L61-L67

Reply:
Thanks a lot for the explanation!